kiali-server/templates/service.yaml

```
@@ -34,6 +34,9 @@ spec:
   {{- end }}
     protocol: TCP
     port: {{ .Values.server.port }}
+    {{- if eq .Values.deployment.service_type "NodePort" }}
```

I haven't tested this but you will want something like this:

```suggestion
    {{- if and (not (empty .Values.server.node_port)) (eq .Values.deployment.service_type "NodePort") }}
```

---

Do you think we should check if it is between **30000-32768** for the port number? I guess you did not check it on the operator side either. I wanted to ask 🤔

---

We don't need to check the value of the node port (we'll let k8s fail if it is outside of its valid range) but we do want to check that node_port is not empty before setting it - this retains the backward compat. behavior.

---

Yes, you are right, I thought so for a moment, but then it didn't make sense to me either, I deleted the message. 😄